src/wms/src/main/java/org/geoserver/wms/featureinfo/GetFeatureInfoKvpReader.java

```diff
@@ -114,23 +181,6 @@ public Object read(Object req, Map kvp, Map rawKvp) throws Exception {
                     + "It should be a proper subset of those instead");
         }
 
-        // remove no-queryable layers
-        for (MapLayerInfo l : request.getQueryLayers().toArray(new MapLayerInfo[0])) {
```

Comment:
Now, this verification is unnecessary, the objects are already filtered in the new GetFeatureInfoKvpRequestReader and GetFeatureInfoKvpParser classes.